python/astro_metadata_translator/translators/decam.py

```diff
@@ -297,3 +298,65 @@ def fix_header(cls, header, instrument, obsid, filename=None):
                       log_label, header["FILTER"], obstype)
 
         return modified
+
+    @classmethod
+    def determine_translatable_headers(cls, filename, primary=None):
+        """Given a file return all the headers usable for metadata translation.
+
+        DECam files are multi-extension FITS with a primary header and
+        each detector stored in a subsequent extension.  DECam uses
+        ``INHERIT=T`` and each detector header will be merged with the
+        primary header.
+
+        Guide headers are not returned.
+
+        Parameters
+        ----------
+        filename : `str`
+            Path to a file in a format understood by this translator.
+        primary : `dict`-like, optional
+            The primary header obtained by the caller. This is sometimes
+            already known, for example if a system is trying to bootstrap
+            without already knowing what data is in the file. Will be
+            merged with detector headers if supplied, else will be read
+            from the file.
+
+        Yields
+        ------
+        headers : iterator of `dict`-like
+            Each detector header in turn.
+
+        Notes
+        -----
+        Each translator class can have code specifically tailored to its
```

Comment:
I think this block might be redundant for subclass methods...

Reply:
modified to say DECam.